### PR TITLE
Add ChipOverBluez build gn compilation support in Linux device layer

### LINKED ISSUE #2441: Add Build gn compilation for ChipOverBluez

### PR DESCRIPTION
**Pull request by yunhanw-google**
Problem
CHIP don't have ChipOverBluez Build gn compilation support

Summary of Changes
-- Add Build gn support for ChipOverBluez
-- Clean unused variables and warning
-- Fix couple of warnings
-- Disable ChipOverBluez support in automake

fixes #2441

### CODE REVIEW

File: gn/chip/linux/gdbus_library.gni [hunk 1]
```diff
@@ -70,6 +70,22 @@ template("gdbus_library") {
       args += [
         "--c-namespace",
         invoker.c_namespace,
+        "--c-namespace",
+        invoker.c_namespace,
```
Comment (mspang) [inline gn/chip/linux/gdbus_library.gni:74]: dupe?

File: src/platform/Linux/CHIPBluezHelper.cpp [hunk 2]
```diff
@@ -277,9 +270,7 @@ static gboolean BluezAdvStop(void * apClosure)
     GDBusObject * adapter;
     BluezEndpoint * endpoint            = static_cast<BluezEndpoint *>(apClosure);
     BluezLEAdvertisingManager1 * advMgr = NULL;
-    GVariantBuilder optionsBuilder;
-    GVariant * options;
-    BluezLEAdvertisement1 * adv;
+    // GVariantBuilder optionsBuilder;
```
Comment (mspang) [inline src/platform/Linux/CHIPBluezHelper.cpp:273]: Remove

File: src/platform/Linux/CHIPBluezHelper.cpp [hunk 3]
```diff
@@ -292,8 +283,8 @@ static gboolean BluezAdvStop(void * apClosure)
     advMgr = bluez_object_get_leadvertising_manager1(BLUEZ_OBJECT(adapter));
     VerifyOrExit(advMgr != NULL, ChipLogProgress(DeviceLayer, "FAIL: NULL advMgr in %s", __func__));
 
-    g_variant_builder_init(&optionsBuilder, G_VARIANT_TYPE("a{sv}"));
-    options = g_variant_builder_end(&optionsBuilder);
+    // g_variant_builder_init(&optionsBuilder, G_VARIANT_TYPE("a{sv}"));
```
Comment (mspang) [inline src/platform/Linux/CHIPBluezHelper.cpp:286]: Remove

File: src/platform/Linux/CHIPBluezHelper.cpp [hunk 4]
```diff
@@ -876,12 +868,11 @@ gboolean BluezPeripheralRegisterApp(void * apClosure)
 
 static void BluezHandleAdvertisementFromDevice(BluezDevice1 * aDevice)
 {
-    const char * address   = bluez_device1_get_address(aDevice);
-    int16_t rssi           = bluez_device1_get_rssi(aDevice);
+    const char * address = bluez_device1_get_address(aDevice);
+    // int16_t rssi           = bluez_device1_get_rssi(aDevice);
```
Comment (mspang) [inline src/platform/Linux/CHIPBluezHelper.cpp:872]: remove

File: src/platform/Linux/CHIPBluezHelper.cpp [hunk 5]
```diff
@@ -987,11 +978,12 @@ static void BluezSignalInterfacePropertiesChanged(GDBusObjectManagerClient * aMa
 
                     data = g_variant_get_fixed_array(value, &len, sizeof(uint8_t));
 
-                    /* TODO why is otPlat API limited to 255 bytes? */
+                    /* TODO why is API limited to 255 bytes? */
                     if (len > 255)
                         len = 255;
 
                     buf = (uint8_t *) g_memdup(data, len);
+                    ChipLogProgress(DeviceLayer, "buf addr is %p", buf);
```
Comment (mspang) [inline src/platform/Linux/CHIPBluezHelper.cpp:986]: Did you mean to land this log?

File: src/platform/Linux/dbus/bluez/BUILD.gn [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import("//build_overrides/chip.gni")
+
+import("${chip_root}/gn/chip/linux/gdbus_library.gni")
+
+gdbus_library("bluez") {
+  sources = [ "org.bluez.xml" ]
+
+  c_namespace = "Bluez"
+  interface_prefix = "org.bluez"
+  c_generate_object_manager = "Yes"
```
Comment (mspang) [inline src/platform/Linux/dbus/bluez/BUILD.gn:24]: Could this be true instead of "Yes" ?

**Review by mspang — CHANGES_REQUESTED** [review]

File: gn/chip/linux/gdbus_library.gni (hunk 1 above)
Comment (yunhanw-google) [reply gn/chip/linux/gdbus_library.gni]: yep, just removed, thanks

File: src/platform/Linux/CHIPBluezHelper.cpp (hunk 3 above)
Comment (yunhanw-google) [reply src/platform/Linux/CHIPBluezHelper.cpp]: yep, just removed, thanks

File: src/platform/Linux/CHIPBluezHelper.cpp (hunk 5 above)
Comment (yunhanw-google) [reply src/platform/Linux/CHIPBluezHelper.cpp]: yep, just removed, thanks

File: src/platform/Linux/dbus/bluez/BUILD.gn (hunk 6 above)
Comment (yunhanw-google) [reply src/platform/Linux/dbus/bluez/BUILD.gn]: Done, thanks

File: src/platform/Linux/CHIPBluezHelper.cpp (hunk 2 above)
Comment (yunhanw-google) [reply src/platform/Linux/CHIPBluezHelper.cpp]: done, thanks

File: src/platform/Linux/CHIPBluezHelper.cpp (hunk 4 above)
Comment (yunhanw-google) [reply src/platform/Linux/CHIPBluezHelper.cpp]: done, thanks

File: src/platform/Linux/dbus/bluez/BUILD.gn [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import("//build_overrides/chip.gni")
+
+import("${chip_root}/gn/chip/linux/gdbus_library.gni")
+
+gdbus_library("bluez") {
+  sources = [ "org.bluez.xml" ]
+
+  c_namespace = "Bluez"
+  interface_prefix = "org.bluez"
+  c_generate_object_manager = "true"
```
Comment (mspang) [inline src/platform/Linux/dbus/bluez/BUILD.gn:24]: boolean true instead of string?

**Review by mspang — APPROVED** [review]

File: src/platform/Linux/dbus/bluez/BUILD.gn (hunk 7 above)
Comment (yunhanw-google) [reply src/platform/Linux/dbus/bluez/BUILD.gn]: Yes, done, thanks

**Review by rwalker-apple — CHANGES_REQUESTED** [review]
@yunhanw-google please update the description using the PR template

**Comment by yunhanw-google** [comment]
> @yunhanw-google please update the description using the PR template

@rwalker-apple Done, thank you!

**Review by rwalker-apple — APPROVED** [review]

**Comment by rwalker-apple** [comment]
@saurabhst ?

**Review by saurabhst — APPROVED** [review]

**Pull request merged**